pkg/storage/watch_cache.go

```diff
@@ -35,9 +35,10 @@ import (
 )
 
 const (
-	// MaximumListWait determines how long we're willing to wait for a
-	// list if a client specified a resource version in the future.
-	MaximumListWait = 60 * time.Second
+	// blockTimeout determines how long we're willing to block the request
+	// to wait for a given resource version to be propagated to cache,
+	// before terminating request and returning BadRequest error.
```

Comment:
I thought about reusing request timeout, but in many cases it's not even set. And it's good to ensure that the request won't be pending forever.

I completely agree with the "retry-after" suggestion. Though I think we have a better solution than 429 - we have "504" (server timeout) that is also setting "retry after".